app/jobs/data_export_xml_job.rb

```diff
@@ -2,7 +2,7 @@ class DataExportXmlJob < ApplicationJob
   queue_as :default
 
   def perform(full_update: false)
-    storage_service = Storage::S3Service.new(Configuration::PaasConfigurationService.new, ENV["EXPORT_PAAS_INSTANCE"])
+    storage_service = Storage::S3Service.new(PlatformHelper.is_paas? ? Configuration::PaasConfigurationService.new : Configuration::EnvConfigurationService.new, ENV["EXPORT_PAAS_INSTANCE"])
```

Comment:
FYI: It may have saved some effort to pull this out into a configuration service used everywhere which handles this check internally. Would solve some of the "in how many places do we do a test" questions too.

Given that this state of affairs only exists until we do the "get rid of paas everywhere" tickets, and that we don't often add configuration-y things, I'm happy to leave it if you think the effort of changing it would be more than an hour or so.

Reply:
Separately, `EXPORT_PAAS_INSTANCE` and similar are (and were already) really badly named things about to become more badly named - I'm going to raise a separate ticket for changing that, although it would be worth us discussing in a refinement session when and how to do it.

Reply:
As agreed, we're not going to refactor the configuration class conditional logic right now.